### PR TITLE
Marge B92 Zephyr updates into new develop branch

### PR DESCRIPTION
**Pull request by serhiiSalamakha**
http://192.168.48.49:8080/browse/MATTER-362

### CODE REVIEW

**Comment by andriy-bilynskyy** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/4cab17cc82b9ca764ffad3df5a621171f7e6f21d (develop,new_develop) | https://github.com/telink-semi/hal_telink/commit/2f4da732d92cf0a785c43f887c3c5ff7d8bb3e68 (telink_B92) | [telink-semi/hal_telink@4cab17cc..2f4da732](https://github.com/telink-semi/hal_telink/compare/4cab17cc82b9ca764ffad3df5a621171f7e6f21d..2f4da732d92cf0a785c43f887c3c5ff7d8bb3e68) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

File: drivers/adc/adc_b9x.c [hunk 1]
```diff
@@ -390,84 +390,77 @@ static int adc_b91_channel_setup(const struct device *dev,
 
 	/* Init ADC */
 	data->differential = channel_cfg->differential;
-	adc_init(vref_internal_mv, pre_scale, sample_freq);
```
Comment (s07641069) [inline drivers/adc/adc_b9x.c:393]: here removed part of code, have we tested impact into B91?

File: drivers/spi/spi_b9x.c [hunk 2]
```diff
@@ -101,55 +111,78 @@ static bool spi_b91_config_cs(const struct device *dev,
...
+			} else {
+				gspi_cs_pin_dis(cs_pin);
+			}
+#endif
 		}
 
 		/* enable cs pin if it is defined and is requested */
 		if ((cs_pin != 0) && (cs_id == config->slave)) {
-			if (b91_config->peripheral_id == PSPI_MODULE) {
-				pspi_set_pin_mux(cs_pin);
```
Comment (s07641069) [inline drivers/spi/spi_b9x.c:114]: Is this code line removed by mistake?> pspi_set_pin_mux(cs_pin);

File: drivers/spi/spi_b9x.c [hunk 3]
```diff
@@ -160,23 +193,18 @@ static void spi_b91_tx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t l
 			tx = 0;
 		}
 		spi_context_update_tx(ctx, 1, 1);
-		while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_TXF_FULL) {
```
Comment (s07641069) [inline drivers/spi/spi_b9x.c:163]: why removed?

File: drivers/spi/spi_b9x.c [hunk 4]
```diff
@@ -160,23 +193,18 @@ static void spi_b91_tx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t l
...
 
 /* process rx data */
 _attribute_ram_code_sec_
-static void spi_b91_rx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t len)
+static void spi_b9x_rx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t len)
 {
 	uint8_t rx = 0;
 
 	for (int i = 0; i < len; i++) {
-		while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_RXF_EMPTY) {
```
Comment (s07641069) [inline drivers/spi/spi_b9x.c:176]: why removed?

File: soc/riscv/riscv-privilege/telink_b9x/Kconfig.defconfig.series [hunk 5]
```diff
@@ -29,7 +29,7 @@ config RISCV_GP
 
 config NUM_IRQS
 	int
-	default 76
+	default 59
```
Comment (s07641069) [inline soc/riscv/riscv-privilege/telink_b9x/Kconfig.defconfig.series:32]: why we need this change?

File: drivers/adc/adc_b9x.c (hunk 1 above)
Comment (serhiiSalamakha) [reply drivers/adc/adc_b9x.c:393]: _**adc_battery_voltage_sample_init**_ and **_adc_gpio_sample_init_** functions call _**adc_init**_.
Yes, it was also tested in B91.

File: drivers/spi/spi_b9x.c (hunk 2 above)
Comment (serhiiSalamakha) [reply drivers/spi/spi_b9x.c:114]: **_pspi_cs_pin_en(cs_pin)_** function is a wrapper for the  **_pspi_set_pin_mux_**. 
It is not necessary to do it twice.
Also **_hspi_cs_pin_en(cs_pin)_** function is a wrapper for the  **_hspi_set_pin_mux_**.

File: drivers/spi/spi_b9x.c (hunk 3 above)
Comment (serhiiSalamakha) [reply drivers/spi/spi_b9x.c:163]: _**hal_spi_write**_(spi_write in hal spi driver) contains the _**while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_TXF_FULL)**_ check.

File: drivers/spi/spi_b9x.c (hunk 4 above)
Comment (serhiiSalamakha) [reply drivers/spi/spi_b9x.c:176]: _**hal_spi_read**_(spi_read in hal spi driver) contains the _**while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_RXF_EMPTY)**_ check.

File: soc/riscv/riscv-privilege/telink_b9x/Kconfig.defconfig.series (hunk 5 above)
Comment (serhiiSalamakha) [reply soc/riscv/riscv-privilege/telink_b9x/Kconfig.defconfig.series:32]: The **_NUM_IRQS_** was updated.

**Review by s07641069 — APPROVED** [review]
@andriy-bilynskyy could you add your review?
Jenkins tests still in progress.
GitHub CI under fixing.

**Review by s07641069 — APPROVED** [review]

File: .github/workflows/telink-build.yaml
Comment (andriy-bilynskyy) [inline .github/workflows/telink-build.yaml:1]: Could you split this job into 2 files for b91 and b92

**Comment by andriy-bilynskyy** [comment]
Could you update both branches to latest develop. In that case I'll start testing our samples from artifacts.

File: .github/workflows/telink-build.yaml
Comment (serhiiSalamakha) [reply .github/workflows/telink-build.yaml:1]: Done

**Review by andriy-bilynskyy — APPROVED** [review]

**Pull request merged**